src/Files.App/ViewModels/UserControls/NavigationToolbarViewModel.cs

```diff
@@ -548,28 +596,124 @@ public void PathBoxItem_PointerPressed(object sender, PointerRoutedEventArgs e)
 			_pointerRoutedEventArgs = ptrPt.Properties.IsMiddleButtonPressed ? e : null;
 		}
 
-		public async Task PathBoxItem_Tapped(object sender, TappedRoutedEventArgs e)
+		public async Task HandleFolderNavigationAsync(string path, bool openNewTab = false)
 		{
-			var itemTappedPath = ((sender as TextBlock)?.DataContext as PathBoxItem)?.Path;
-			if (itemTappedPath is null)
-				return;
-
-			if (_pointerRoutedEventArgs is not null)
+			openNewTab |= _pointerRoutedEventArgs is not null;
+			if (openNewTab)
 			{
-				await MainWindow.Instance.DispatcherQueue.EnqueueOrInvokeAsync(async () =>
-				{
-					await NavigationHelpers.AddNewTabByPathAsync(typeof(ShellPanesPage), itemTappedPath, true);
-				}, DispatcherQueuePriority.Low);
-				e.Handled = true;
+				await MainWindow.Instance.DispatcherQueue.EnqueueOrInvokeAsync(
+					async () =>
+					{
+						await NavigationHelpers.AddNewTabByPathAsync(typeof(ShellPanesPage), path, true);
+					},
+					DispatcherQueuePriority.Low);
+
 				_pointerRoutedEventArgs = null;
 
 				return;
 			}
 
-			ToolbarPathItemInvoked?.Invoke(this, new PathNavigationEventArgs()
+			ToolbarPathItemInvoked?.Invoke(this, new() { ItemPath = path });
+		}
+
+		public async Task HandleItemNavigationAsync(string path)
```

Comment:
Where is all this logic taken from?

Reply:
From here:

https://github.com/files-community/Files/blob/15e864a85e664265c3e16d03354be13b07701273/src/Files.App/ViewModels/UserControls/NavigationToolbarViewModel.cs#L758

Reply:
Are there any modifications?